src/main.ts

```
@@ -484,6 +503,72 @@ export class SouthernCompanyAPI extends EventEmitter{
 		return monthlyData;
 	}
 
+	public async getHourlyData(startDate: Date, endDate: Date){
+		let accounts = this.getAccountsArray();
+
+		const hourlyDataReponse = await Promise.all(accounts.map(account => this.buildHourlyDataResponse(account, startDate, endDate)))
+
+		return hourlyDataReponse
+	}
+
+	private async buildHourlyDataResponse(account: string, startDate: Date, endDate: Date) {
+		const servicePointNumber = await this.fetchServicePointNumber(account)
+		const url = this.buildHourlyURL(startDate, endDate, account, servicePointNumber);
+
+		const response = await fetch(url, {
+			method: 'GET',
+			headers: {
+				'Content-Type': 'application/json; charset=UTF-8',
+				Authorization: `bearer ${this.jwt}`
+			}
+		});
+
+		if (response.status !== 200) {
+			throw new Error(`Failed to get hourly data: ${response.statusText} with endpoint: ${url}`);
+		}
+
+		const jsonResponse: API.hourlyMPUData = await response.json();
+
+		const graphData: API.hourlyMPUGraphData = JSON.parse(jsonResponse.Data.Data)
+
+		let combinedGraphData
+
+		if (graphData) {
```

What happens if we don't have graphData? When does this occur? Should we throw an error if it is falsey?

---

In my testing, this occurs when you select an invalid date range, e.g. From Today - Today OR if you select a date range that is within the N - 2 limitation. There are similar patterns elsewhere in the code, so I mostly followed those